frontend/src/components/LoginPage.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { RootSchema } from "../store/root-schema";
+import { connect } from "react-redux";
+import { Dispatch } from "../store";
+import { connectWebSocket, setUsername } from "../store/actions";
+
+interface Props {
+  username: string;
+  connecting: boolean;
+  dispatch: Dispatch;
+}
+
+const LoginPage: React.FunctionComponent<Props> = ({
+  username,
+  connecting,
+  dispatch,
+}: Props) => {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    if (connecting) return;
+
+    dispatch(setUsername(e.target.value));
```

Comment:
I don't think that saving every inputted letter is a good idea. We should save it when user clicks submit so with `connectWebSocket` action.